Automattic/wp-calypso
# Add reviews trigger button

Closes https://github.com/Automattic/dotcom-forge/issues/4560

## Proposed Changes

* add the `marketplace-reviews-show` feature flag
* add the trigger button with the number of reviews
* only show the button when the feature flag is enabled 

## Testing Instructions



* Visit a plugin that already has reviews, etc. `/plugins/woocommerce-bookings`
* Check that the button is rendered on the page
* Disable the feature flag, by adding the `?flags=-marketplace-reviews-show` param to the URL
* Refresh the page and check that the button is not rendered anymore
* Enable the feature flag again, by adding the `?flags=marketplace-reviews-show` param to the URL
* Visit a plugin page that has no reviews
* Check that the button is not rendered on the page

| `marketplace-reviews-show` feature flag | `numberOfReviews` > 0 | Result |
|--------|--------|--------|
| `true` | `true`| ![CleanShot 2023-11-24 at 11 36 33@2x](https://github.com/Automattic/wp-calypso/assets/11555574/dc0a2071-2d8c-41b6-b1cd-bed5fa00844a) |
| `true` | `false` | ![CleanShot 2023-11-24 at 11 37 10@2x](https://github.com/Automattic/wp-calypso/assets/11555574/cb98de7a-29a8-4fdb-891c-c153ba817547) |
| `false` | `true` | ![CleanShot 2023-11-24 at 11 36 21@2x](https://github.com/Automattic/wp-calypso/assets/11555574/77978b7d-123b-421a-a838-dcbf8207cc14) | 
| `false` | `false` | ![CleanShot 2023-11-24 at 11 37 10@2x](https://github.com/Automattic/wp-calypso/assets/11555574/cb98de7a-29a8-4fdb-891c-c153ba817547) | 

## Pre-merge Checklist
